File: app/src/main/java/za/co/jpsoft/winkerkreader/MyService.java
package za.co.jpsoft.winkerkreader;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Handler;
import android.os.IBinder;
import android.telephony.PhoneStateListener;
import android.telephony.TelephonyManager;
import android.widget.Toast;
import static za.co.jpsoft.winkerkreader.data.WinkerkContract.*;
public class MyService extends Service {

	@Override
	public IBinder onBind(Intent intent) {

		return null;
	}

	@Override
	public void onCreate() {
		Toast.makeText(getBaseContext(),
				"Application Is Running in Background", Toast.LENGTH_SHORT)
				.show();
		boolean isOn = true;
		final SharedPreferences settings = getSharedPreferences(PREFS_USER_INFO, 0);
		final SharedPreferences.Editor editor = settings.edit();
		TelephonyManager telephonyManager = (TelephonyManager) getSystemService(Context.TELEPHONY_SERVICE);
		PhoneStateListener phoneStateListener = new PhoneStateListener() {
			@Override
			public void onCallStateChanged(int state, String incomingNumber) {
				if (state == TelephonyManager.CALL_STATE_RINGING) {
					if (incomingNumber != null) {
						editor.putString("CallerNumber", incomingNumber);} else {
						editor.putString("CallerNumber", "Unknown");}
					editor.commit();
					startService(new Intent(getApplicationContext(), oproepdetail.class));
				} else if (oproepdetail.isOn) {
					final Handler handler = new Handler();
					handler.postDelayed(new Runnable() {
						@Override
						public void run() {
							editor.putString("CallerNumber", "XXXXXXXXXX");
							editor.apply();
							stopService(new Intent(getApplicationContext(), oproepdetail.class));
						}
					}, 10000);
				}
			}
		};
		telephonyManager.listen(phoneStateListener, PhoneStateListener.LISTEN_CALL_STATE);

		super.onCreate();
	}

	@Override
	public int onStartCommand(Intent intent, int flags, int startId) {
		// TODO Auto-generated method stub
		return super.onStartCommand(intent, flags, startId);
	}

}

File: app/src/main/kotlin/za/co/jpsoft/winkerkreader/services/MyService.kt
package za.co.jpsoft.winkerkreader.services


import android.app.Service
import android.content.Intent
import android.os.Handler
import android.os.IBinder
import android.telephony.PhoneStateListener
import android.telephony.TelephonyManager
import android.widget.Toast
import za.co.jpsoft.winkerkreader.data.WinkerkContract.PREFS_USER_INFO

class MyService : Service() {

    override fun onBind(intent: Intent): IBinder? = null

    override fun onCreate() {
        super.onCreate()

        Toast.makeText(
            baseContext,
            "Application Is Running in Background",
            Toast.LENGTH_SHORT
        ).show()

        val settings = getSharedPreferences(PREFS_USER_INFO, 0)
        val editor = settings.edit()

        val telephonyManager = getSystemService(TELEPHONY_SERVICE) as TelephonyManager

        val phoneStateListener = object : PhoneStateListener() {
            override fun onCallStateChanged(state: Int, incomingNumber: String?) {
                when (state) {
                    TelephonyManager.CALL_STATE_RINGING -> {
                        editor.putString("CallerNumber", incomingNumber ?: "Unknown").apply()
                        startService(Intent(applicationContext, OproepDetailService::class.java))
                    }
                    else -> {
                        if (OproepDetailService.isOn) {
                            Handler().postDelayed({
                                editor.putString("CallerNumber", "XXXXXXXXXX").apply()
                                stopService(Intent(applicationContext, OproepDetailService::class.java))
                            }, 10000)
                        }
                    }
                }
            }
        }

        telephonyManager.listen(phoneStateListener, PhoneStateListener.LISTEN_CALL_STATE)
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        return super.onStartCommand(intent, flags, startId)
    }
}